llvm/lib/Target/AIE/AIEHazardRecognizer.cpp

```diff
@@ -121,6 +121,9 @@ bool AIEResourceCycle::canReserveResources(MachineInstr &MI) {
   if (!AlternateInsts)
     return Bundle.canAdd(&MI);
 
+  if (MI.mayLoad())
```

Comment:
Can you add a comment to explain that change? So far we want the change for VLD pseudos to be NFC for the SW pipeliner